### PR TITLE
Query and Update Animals table

### PR DESCRIPTION
**Pull request by Obimbo07**
This pull request introduces the following changes to the database i have attached screenshots to show results of actions:

- Inside a transaction update the animals table by setting the species column to unspecified.

284593368-514c8b76-215e-478f-a76f-1c9b0bb6b574

- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Capture3

-  Inside a transaction delete all records in the animals table, then roll back the transaction.
Capture4

- Delete all animals born after Jan 1st, 2022.
- Create a savepoint for the transaction.
Update all animals' weight to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction
Capture5

Below are queries to answer this questions: 

1. How many animals are there?
Capture6

2.  How many animals have never tried to escape?
Capture7
3. What is the average weight of animals?
Capture9
4. Who escapes the most, neutered or not neutered animals?
Capture10
5. What is the minimum and maximum weight of each type of animal.
Capture116.
6. What is the average number of escape attempts per animal type of those born between 1990 and 2000?
Capture12

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -12,4 +12,55 @@ SELECT * FROM animals WHERE neutered = true;
 
 SELECT * FROM animals WHERE name != 'Gabumon';
 
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;   // const user = userData ? JSON.parse(userData) : null
+
+
+ALTER TABLE animals ADD COLUMN species VARCHAR(100);
```
Comment (shahadat3669) [inline queries.sql]: 👏 Good job on adding the new `species` column with a string type! 👏 However, there's a small adjustment needed:

- [ ] **Organizing Queries in `schema.sql`:** According to project requirements, any modifications to the table structure should be saved in the `schema.sql` file. Please consider moving the `ALTER` query for the new `species` column from `queries.sql` to the appropriate section in the `schema.sql` file.

By ensuring the query is in the correct file, you'll maintain consistency with project guidelines. Keep up the great work! 🚀📄

File: queries.sql [hunk 2]
```diff
@@ -12,4 +12,55 @@ SELECT * FROM animals WHERE neutered = true;
...
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
```
Comment (shahadat3669) [inline queries.sql:22]: 👏 Great job on implementing the first three parts of the transaction! 👏 However, according to the project requirements, there is one more step needed after the rollback. Let's address this to ensure completion:

- [ ] **Verification After Rollback:** After rolling back the transaction, it's essential to verify that the `species` column has returned to its state before the transaction. Please add this fourth step to complete the transaction process.
- [ ] Additionally, please consider to add the screenshot in the PR description too.

By incorporating this verification step, your implementation will align perfectly with the project requirements. Keep up the excellent work! 🚀🔍

File: queries.sql [hunk 3]
```diff
@@ -12,4 +12,55 @@ SELECT * FROM animals WHERE neutered = true;
...
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
```
Comment (shahadat3669) [inline queries.sql:30]: 👏 Great progress on updating the `species` column in the `animals` table! 👏 However, to align with the project requirements, there's an additional step needed after the commit. Let's address this to ensure completeness:

- [ ] **Verification After Commit:** After committing the changes to the `animals` table, it's crucial to re-verify, just as you did before the commit. Please add the query to confirm that the changes persist in the database after the commit.
image
- [ ] Additionally, please consider to add the screenshot in the PR description too.
By including this verification step, your implementation will fully meet the project requirements. Keep up the good work! 🚀🔍

File: queries.sql [hunk 4]
```diff
@@ -12,4 +12,55 @@ SELECT * FROM animals WHERE neutered = true;
...
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT before_weight_update;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO before_weight_update;
+UPDATE animals SET weight_kg = ABS(weight_kg) WHERE weight_kg < 0;
```
Comment (shahadat3669) [inline queries.sql]: 👏 You've made a comprehensive analysis of the statement! 👏 However, there's a nuanced difference between your query and the corrected statement. Let's delve into it:

- [ ] **Preserving Negative Sign in Weights:** Your query utilizes the `ABS()` function, which takes the absolute value of the weight, effectively removing the negative sign. However, the project requirement requested to use multiplication by -1, aligning with the project requirement to change the sign to positive using this method.
- [ ] Additionally if necessary please consider updating the Screenshots in the PR Description.

By understanding this distinction, you can choose the approach that best aligns with the specific project requirement. Keep up the meticulous review! 🚀🔍

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @Obimbo07, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ add a new column called `species` with a string type
✅ add the requested data
✅ Added Screenshots in the PR Description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Obimbo07, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**